src/llscan.h

```diff
@@ -174,36 +181,75 @@ class TypeRecord {
 
 
  private:
+  friend DetailedTypeRecord;
   std::string type_name_;
   uint64_t instance_count_;
   uint64_t total_instance_size_;
   std::set<uint64_t> instances_;
 };
 
+class DetailedTypeRecord : public TypeRecord {
+ public:
+  DetailedTypeRecord(std::string& type_name, uint64_t own_descriptors_count,
+                     uint64_t indexed_properties_count)
+      : TypeRecord(type_name),
+        own_descriptors_count_(own_descriptors_count),
+        indexed_properties_count_(indexed_properties_count) {}
+  const uint64_t GetOwnDescriptorsCount() { return own_descriptors_count_; };
```

Comment:
`uint6_t own_descriptors_count() const { ... }` since it's a simple getter and note that the `const` should come after the method name.  Putting it on the type is a no-op and one that some compilers complain about.

edit: oh, I see there are some existing baroque getter names.  Not wrong then.